com.ibm.jbatch.tck/src/main/java/com/ibm/jbatch/tck/artifacts/specialized/PUDPartitionReader.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2016 International Business Machines Corp.
+ * 
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Licensed under the Apache License, 
+ * Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+package com.ibm.jbatch.tck.artifacts.specialized;
+
+import java.io.Serializable;
+
+import javax.batch.api.BatchProperty;
+import javax.batch.api.chunk.AbstractItemReader;
+import javax.batch.runtime.context.JobContext;
+import javax.batch.runtime.context.StepContext;
+import javax.inject.Inject;
+
+import com.ibm.jbatch.tck.artifacts.basicchunk.BasicItem;
+
+/*NOTE: Code for this class is taken substantially from basicchunk.BasicReader*/
+@javax.inject.Named("PUDPartitionReader")
+public class PUDPartitionReader extends AbstractItemReader {
+
+    @Inject 
+    JobContext jobCtx;
+    
+    @Inject 
+    StepContext stepCtx;
+    
+    @Inject @BatchProperty(name="partition.number")
+    String partitionNumber;
+    
+    @Inject @BatchProperty(name="execution.number")
+    String executionNumber;
+    
+    @Inject    
+    @BatchProperty(name = "number.of.items.to.be.read")
+    String injectedNumberOfItemsToBeRead;
+    //Default: read 10 items
+    private int numberOfItemsToBeRead = 10;
+    
+    @Inject
+    @BatchProperty(name = "throw.reader.exception.for.these.items")
+    String injectedThrowReaderExceptionForTheseItems;
+    //Default: don't throw any exceptions
+    private int[] throwReaderExceptionForTheseItems = {};
+
+    private String PUDString;
+    private int currentItemId = -1;
+    private BasicItem currentItem = null;
+    
+    @Override
+    public void open(Serializable checkpoint) {
+    	
+        PUDString = "PUD for Partition: " +partitionNumber;
+        
+        // Set on the first execution; on later executions it will need to be obtained
+        // from the job repository's persistent store.
+        if (checkpoint == null) {
+            stepCtx.setPersistentUserData(PUDString);
+        }
+        
```

Comment:
I changed this since if you're going to set it every time doesn't it render the check in readItem() less interesting in exec #2?